<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>chromatic in scATAC-seq • chromatic</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Montserrat-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="chromatic in scATAC-seq">
<meta name="description" content="Tutorial for calculating chromatin entropy in scATAC-seq data
">
<meta property="og:description" content="Tutorial for calculating chromatin entropy in scATAC-seq data
">
<link rel="stylesheet" href="extra.css">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">chromatic</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.03</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/scATAC.html">scATAC-seq tutorial</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changes</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/smorabit/chromatic/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>chromatic in scATAC-seq</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/smorabit/chromatic/blob/main/vignettes/scATAC.Rmd" class="external-link"><code>vignettes/scATAC.Rmd</code></a></small>
      <div class="d-none name"><code>scATAC.Rmd</code></div>
    </div>

    
    
<p>Compiled: 27-09-2025</p>
<p>Source: <code>vignettes/scATAC.Rmd</code></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This tutorial demonstrates the key functionality of
<code>chromatic</code> to calculate cell-level <em>chromatin
entropy</em> and <em>chromatin erosion</em> scores in a scATAC-seq
dataset. For this tutorial, we demonstrate the package on a dataset of
colorectal cancer tumors profiled with 10X Genomics Multiome (ATAC +
RNA), although only the ATAC assay is needed for the analysis. At this
time the dataset is not currently available to the public, but in the
future we will update this tutorial to use a publicly available dataset.
To follow along with this tutorial, please use your own dataset. First
let’s define these scores.</p>
<ul>
<li><p><strong>Entropy score</strong>: Measures the diversity of
chromatin states within a cell, with higher values indicating greater
epigenomic heterogeneity or plasticity.</p></li>
<li><p><strong>Erosion score</strong>: Quantifies the degree to which a
cell’s chromatin state profile shifts toward repressive versus active
states, based on signed and normalized chromatin state abundances.
Positive scores indicate an enrichment of repressive states while
negative values indicate enrichment of active states.</p></li>
</ul>
<hr>
<div class="section level3">
<h3 id="load-the-dataset-and-required-libraries">Load the dataset and required libraries<a class="anchor" aria-label="anchor" href="#load-the-dataset-and-required-libraries"></a>
</h3>
<p>First we load the processed dataset and the required R libraries for
this tutorial.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># single-cell analysis packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/stuart-lab/signac" class="external-link">Signac</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># data / plotting packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="fu">theme_set</span><span class="op">(</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_cowplot.html" class="external-link">theme_cowplot</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># additional Genomics packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v86</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># load the processed Seurat object</span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"CRC_Tumor_multiome.rds"</span><span class="op">)</span></span></code></pre></div>
<p>To get acquainted with this dataset, let’s check the UMAP colored by
clusters.</p>
<p><img src="figures/scATAC/crc_umap.png" width="700" height="700"></p>
<details><summary>
See code
</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">'anno'</span>,</span>
<span>  reduction <span class="op">=</span> <span class="st">'umap'</span>,</span>
<span>  label <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  repel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  raster<span class="op">=</span><span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_void</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">""</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span></span>
<span>    panel.border <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>fill<span class="op">=</span><span class="cn">NA</span>, color<span class="op">=</span><span class="st">'black'</span>, linewidth<span class="op">=</span><span class="fl">0.25</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</details><hr>
</div>
</div>
<div class="section level2">
<h2 id="run-chromatic">Run chromatic<a class="anchor" aria-label="anchor" href="#run-chromatic"></a>
</h2>
<div class="section level3">
<h3 id="set-up-chromatin-states">Set up chromatin states<a class="anchor" aria-label="anchor" href="#set-up-chromatin-states"></a>
</h3>
<p>In order to run <code>chromatic</code>, we require a <a href="https://bioconductor.org/packages/devel/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html" class="external-link"><code>GenomicRanges</code></a>
object containing information about which genomic regions correspond to
which chromatin state. We recommend using chromatin states generated via
<a href="https://compbio.mit.edu/ChromHMM/" class="external-link"><code>ChromHMM</code></a>.</p>
<p>Importantly, chromatin states exhibit considerable biological
variation across cell lineages, tissues, and species. Therefore it is
important to choose an appropriate reference for our dataset. In this
tutorial, we are using a colorectal cancer dataset, so we require a
chromatin state reference for the colon.</p>
<p>The <a href="https://egg2.wustl.edu/roadmap/web_portal/index.html" class="external-link">ROADMAP
epigenomics project</a> hosts chromatin states that have been computed
in many different human tissues, including the colon. We use the
function <code>FetchChromatinStates</code> to download the relevant
ChromHMM chromatin states, and load it into R as a
<code>GenomicRanges</code> object.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">chromHMM_states</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FetchChromatinStates.html">FetchChromatinStates</a></span><span class="op">(</span></span>
<span>  tissue <span class="op">=</span> <span class="st">"Colon"</span>, <span class="co"># input the tissue ID, mnemonic, or name</span></span>
<span>  model <span class="op">=</span> <span class="fl">15</span>, <span class="co"># select the 15 state, 18 state, or 25 state model</span></span>
<span>  genome <span class="op">=</span> <span class="st">'hg38'</span> <span class="co"># select hg19 or hg38</span></span>
<span><span class="op">)</span></span></code></pre></div>
<details><summary>
Additional information
</summary><p>When we run this function, first we see the following message:</p>
<pre><code>Multiple tissues matched your input:
   EID      Mnemonic                Name
1 E076 GI.CLN.SM.MUS Colon Smooth Muscle
2 E106    GI.CLN.SIG       Sigmoid Colon
3 E075    GI.CLN.MUC      Colonic Mucosa
Enter the row number of the tissue you want: 2

Downloading ChromHMM segmentation from: https://egg2.wustl.edu/roadmap/data/byFileType/chromhmmSegmentations/ChmmModels/coreMarks/jointModel/final/E106_15_coreMarks_hg38lift_mnemonics.bed.gz
trying URL 'https://egg2.wustl.edu/roadmap/data/byFileType/chromhmmSegmentations/ChmmModels/coreMarks/jointModel/final/E106_15_coreMarks_hg38lift_mnemonics.bed.gz'
Content type 'application/x-gzip' length 2302510 bytes (2.2 MB)
==================================================
downloaded 2.2 MB

Loaded 393420 regions for E106 (GI.CLN.SIG, Sigmoid Colon) — 15-state model, genome hg38</code></pre>
<p>The function tries to match the <code>tissue</code> argument to the
list of tissues profiled in the ROADMAP project. The table containing
this information is included in <code>chromatic</code> in an object
called <code>tissue_map</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tissue_map</span><span class="op">)</span></span></code></pre></div>
<pre><code>head(tissue_map)
   EID GROUP   Mnemonic                                   Name
1 E017 IMR90  LNG.IMR90 IMR90 fetal lung fibroblasts Cell Line
2 E002   ESC    ESC.WA7                           ES-WA7 Cells
3 E008   ESC     ESC.H9                               H9 Cells
4 E001   ESC     ESC.I3                            ES-I3 Cells
5 E015   ESC  ESC.HUES6                            HUES6 Cells
6 E014   ESC ESC.HUES48                           HUES48 Cells</code></pre>
<p>For convenience, <code>FetchChromatinStates</code> accepts partial
matches, and prompts the user to select the best match for their tissue.
Partial matching can be turned off by setting
<code>fuzzy_matching=FALSE</code>.</p>
</details><p>Let’s inspect the output of this function.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">chromHMM_states</span><span class="op">)</span></span></code></pre></div>
<pre><code>GRanges object with 6 ranges and 1 metadata column:
      seqnames        ranges strand |        name
         &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
  [1]     chr1   10001-14600      * |    15_Quies
  [2]     chr1   14601-19000      * |      5_TxWk
  [3]     chr1   19001-96080      * |    15_Quies
  [4]     chr1   96277-96476      * |    15_Quies
  [5]     chr1  97277-177200      * |    15_Quies
  [6]     chr1 257850-297849      * |    15_Quies
  -------
  seqinfo: 25 sequences from an unspecified genome; no seqlengths</code></pre>
<p>If you are using an external resource aside from ROADMAP, for example
if you are using a non-human species or a tissue not profiled by
ROADMAP, you must format your data as a <code>GenomicRanges</code>
object similar to what is shown above.</p>
<hr>
</div>
<div class="section level3">
<h3 id="runchromatic">
<code>RunChromatic</code><a class="anchor" aria-label="anchor" href="#runchromatic"></a>
</h3>
<p>The main function of <code>chromatic</code> is
<code>RunChromatic</code>, which will calculate a chromatin erosion
score an a chromatin entropy score for each cell in the
<code>seurat_obj</code>. These scores are based on matching regions in a
<a href="https://stuartlab.org/signac/reference/createchromatinassay?search-input=chromatinassay" class="external-link"><code>ChromatinAssay</code></a>
in our <code>seurat_obj</code> with the regions from our chromatin
states from the previous step.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RunChromatic.html">RunChromatic</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>, </span>
<span>  <span class="va">chromHMM_states</span>,</span>
<span>  stoplist <span class="op">=</span> <span class="va">blacklist_hg38_unified</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">'Peaks'</span>,</span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add the scores to the seurat_obj@meta.data slot</span></span>
<span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">entropy_score</span> <span class="op">&lt;-</span> <span class="va">output</span><span class="op">$</span><span class="va">entropy</span><span class="op">$</span><span class="va">entropy</span></span>
<span><span class="va">seurat_obj</span><span class="op">$</span><span class="va">erosion_score</span> <span class="op">&lt;-</span> <span class="va">output</span><span class="op">$</span><span class="va">erosion</span><span class="op">$</span><span class="va">erosion</span></span></code></pre></div>
<details><summary>
See function messages
</summary><pre><code>[1] "Filtering by stoplist regions"
[1] "Filtering nonstandard chromosomes"
[1] "Annotating peaks by overlapping with chromatin states"
[1] "Excluding uncommon peaks"
Filtered peaks: retained 158958 of 175508 peaks (90.6%).
[1] "State matrix"
The following states were not classified (sign=0): 10_TssBiv, 8_ZNF/Rpts, 7_Enh, 12_EnhBiv, 11_BivFlnk
[1] "Calculating erosion score "
[1] "Calculating entropy score "</code></pre>
</details><p>The <code>output</code> object is a list containing sevral different
results:</p>
<ul>
<li>
<code>peaks_gr</code>, a subset of the <code>GenomicRanges</code>
object from the <code>ChromatinAssay</code> of your Seurat object, with
the overlapping chromatin state annotation.</li>
<li>
<code>state_matrix</code>, the cells by chromatin states counts
matrix used for the entropy and erosion score calculations.</li>
<li>
<code>entropy</code>, a dataframe containing the entropy score
results.</li>
<li>
<code>erosion</code>, a dataframe containing the erosion score
results.</li>
</ul>
<p>For now we just focus on the entropy and erosion scores.</p>
<hr>
</div>
<div class="section level3">
<h3 id="downstream-plotting">Downstream plotting<a class="anchor" aria-label="anchor" href="#downstream-plotting"></a>
</h3>
<p>Currently, <code>chromatic</code> does not include plotting
functions, and instead we recommend visualizing the results directly
using <code>ggplot2</code>.</p>
<p>First, let’s visualize the distribution of these scores across our
clusters.</p>
<p><img src="figures/scATAC/crc_entropy_distributions.png" width="700" height="700"></p>
<details><summary>
See plotting code
</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">lv2</span>, y <span class="op">=</span> <span class="va">entropy_score</span>, color <span class="op">=</span> <span class="va">anno</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot the points on the bottom:</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggrastr</span><span class="fu">::</span><span class="fu">rasterise</span><span class="op">(</span><span class="fu">ggbeeswarm</span><span class="fu">::</span><span class="fu">geom_quasirandom</span><span class="op">(</span></span>
<span>        method <span class="op">=</span> <span class="st">"pseudorandom"</span>,</span>
<span>        size <span class="op">=</span> <span class="fl">1</span></span>
<span>    <span class="op">)</span>, dpi<span class="op">=</span><span class="fl">300</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add the box </span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> </span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span></span>
<span>        color <span class="op">=</span> <span class="st">'black'</span>,</span>
<span>        outlier.shape<span class="op">=</span><span class="cn">NA</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>        width <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>        fill <span class="op">=</span> <span class="cn">NA</span></span>
<span>    <span class="op">)</span> </span>
<span></span>
<span><span class="co"># set up the theme:</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> </span>
<span>    <span class="fu">theme</span><span class="op">(</span></span>
<span>        axis.line.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.line.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        panel.border <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>linewidth<span class="op">=</span><span class="fl">1</span>,color<span class="op">=</span><span class="st">'black'</span>, fill<span class="op">=</span><span class="cn">NA</span><span class="op">)</span>,</span>
<span>        panel.grid.major.y <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>linewidth<span class="op">=</span><span class="fl">0.25</span>, color<span class="op">=</span><span class="st">'lightgrey'</span><span class="op">)</span>,</span>
<span>        plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        strip.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        strip.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face<span class="op">=</span><span class="st">'bold'</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">RotatedAxis</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">xlab</span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"Entropy score"</span><span class="op">)</span>  <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Chromatin entropy by cluster"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
</details><p><img src="figures/scATAC/crc_erosion_distributions.png" width="700" height="700"></p>
<details><summary>
See plotting code
</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">lv2</span>, y <span class="op">=</span> <span class="va">erosion_score</span>, color <span class="op">=</span> <span class="va">anno</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot the points on the bottom:</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggrastr</span><span class="fu">::</span><span class="fu">rasterise</span><span class="op">(</span><span class="fu">ggbeeswarm</span><span class="fu">::</span><span class="fu">geom_quasirandom</span><span class="op">(</span></span>
<span>        method <span class="op">=</span> <span class="st">"pseudorandom"</span>,</span>
<span>        size <span class="op">=</span> <span class="fl">1</span></span>
<span>    <span class="op">)</span>, dpi<span class="op">=</span><span class="fl">300</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># add the box </span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> </span>
<span>    <span class="fu">geom_boxplot</span><span class="op">(</span></span>
<span>        color <span class="op">=</span> <span class="st">'black'</span>,</span>
<span>        outlier.shape<span class="op">=</span><span class="cn">NA</span>, </span>
<span>        alpha <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>        width <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>        fill <span class="op">=</span> <span class="cn">NA</span></span>
<span>    <span class="op">)</span> </span>
<span></span>
<span><span class="co"># set up the theme:</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> </span>
<span>    <span class="fu">theme</span><span class="op">(</span></span>
<span>        axis.line.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.line.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        panel.border <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>linewidth<span class="op">=</span><span class="fl">1</span>,color<span class="op">=</span><span class="st">'black'</span>, fill<span class="op">=</span><span class="cn">NA</span><span class="op">)</span>,</span>
<span>        panel.grid.major.y <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>linewidth<span class="op">=</span><span class="fl">0.25</span>, color<span class="op">=</span><span class="st">'lightgrey'</span><span class="op">)</span>,</span>
<span>        plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        strip.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>        strip.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face<span class="op">=</span><span class="st">'bold'</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">RotatedAxis</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">xlab</span><span class="op">(</span><span class="st">''</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ylab</span><span class="op">(</span><span class="st">"Erosion score"</span><span class="op">)</span>  <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Chromatin erosion by cluster"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
</details><p>We can also directly compare the entropy and the erosion scores on a
scatter plot.</p>
<p><img src="figures/scATAC/crc_erosion_v_entropy.png" width="700" height="700"></p>
<details><summary>
See plotting code
</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">erosion_score</span>, y <span class="op">=</span> <span class="va">entropy_score</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">ggrastr</span><span class="fu">::</span><span class="fu">rasterise</span><span class="op">(</span><span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>color<span class="op">=</span><span class="va">anno</span><span class="op">)</span>, size<span class="op">=</span><span class="fl">1</span><span class="op">)</span>, dpi<span class="op">=</span><span class="fl">300</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggpubr</span><span class="fu">::</span><span class="fu">stat_cor</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_smooth</span><span class="op">(</span>method <span class="op">=</span> <span class="st">'lm'</span>, color<span class="op">=</span><span class="st">'black'</span><span class="op">)</span>  <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span></span>
<span>      axis.line.x <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      axis.line.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      panel.border <span class="op">=</span> <span class="fu">element_rect</span><span class="op">(</span>linewidth<span class="op">=</span><span class="fl">1</span>,color<span class="op">=</span><span class="st">'black'</span>, fill<span class="op">=</span><span class="cn">NA</span><span class="op">)</span>,</span>
<span>      panel.grid.major <span class="op">=</span> <span class="fu">element_line</span><span class="op">(</span>linewidth<span class="op">=</span><span class="fl">0.25</span>, color<span class="op">=</span><span class="st">'lightgrey'</span><span class="op">)</span>,</span>
<span>      plot.title <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>hjust<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span>,</span>
<span>      strip.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,</span>
<span>      strip.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>face<span class="op">=</span><span class="st">'bold'</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span>   <span class="fu">xlab</span><span class="op">(</span><span class="st">"Erosion score"</span><span class="op">)</span>  <span class="op">+</span></span>
<span>   <span class="fu">ylab</span><span class="op">(</span><span class="st">"Entropy score"</span><span class="op">)</span> <span class="op">+</span></span>
<span>   <span class="fu">labs</span><span class="op">(</span>color <span class="op">=</span> <span class="st">""</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Entropy vs. Erosion"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span></span></code></pre></div>
</details><p>We can also visualize the scores directly on the UMAP, although this
is arguably more difficult than just looking at the distributions due to
cells plotting on top of one another as you can see in the Entropy plot
below.</p>
<p><img src="figures/scATAC/crc_umap_erosion.png" width="700" height="700"></p>
<details><summary>
See plotting code
</summary><p><em>WARNING</em>: The function below is a custom helper function that
I have not included in this package so you sadly will not be able to run
this. I recommend using Seurat <code>FeaturePlot</code> instead.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">FeatureEmbedding</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'erosion_score'</span><span class="op">)</span> ,</span>
<span>  raster<span class="op">=</span><span class="cn">TRUE</span>, dpi<span class="op">=</span><span class="fl">300</span>,</span>
<span>  reduction <span class="op">=</span> <span class="st">'umap'</span>,</span>
<span>  ncol <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  plot_max <span class="op">=</span> <span class="st">'q100'</span>,</span>
<span>  plot_min <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>color<span class="op">=</span><span class="st">"Erosion"</span><span class="op">)</span> </span>
<span></span>
<span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">FeatureEmbedding</span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'entropy_score'</span><span class="op">)</span> ,</span>
<span>  raster<span class="op">=</span><span class="cn">TRUE</span>, dpi<span class="op">=</span><span class="fl">300</span>,</span>
<span>  reduction <span class="op">=</span> <span class="st">'umap'</span>,</span>
<span>  ncol <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  plot_max <span class="op">=</span> <span class="st">'q100'</span>,</span>
<span>  plot_min <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>color<span class="op">=</span><span class="st">"Entropy"</span><span class="op">)</span> </span>
<span></span>
<span><span class="op">(</span><span class="va">p1</span> <span class="op">|</span> <span class="va">p2</span><span class="op">)</span> </span></code></pre></div>
</details><p>Overall, these plots show us that the cells in our dataset display a
range of entropy and erosion values, and that the cells belonging to
malignant clusters usually higher entropy and erosion compared to
non-malignant cells. We also note that there is a strong correlation
between entropy and erosion, but ultimately they are two different yet
complementary measurements of cell-level chromatin heterogeneity.</p>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This concludes the tutorial for running <code>chromatic</code> on a
scATAC-seq dataset in the alpha version of the package. We plan to
expand the tutorial with more features and downstream analysis in future
updates.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://smorabit.github.io" class="external-link">Sam Morabito</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
